### ISSUE TITLE
Update Process Bookkeeping

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind_project#14]
Per #10 and #8 and several discussions today.

We need to update Lind so the Repy side can keep track of separate processes. This will allow us to safely handle files between processes and implement pipes. 

To do this we need to:

1. Update the RPC calls in NaCl so they transmit the Cage-ID. These will need to be traced back to where we have `nap` access in `nacl_syscall_common.c`, and the RPC needs to be modified in `lind_platform.c`
2. Modify the `fork` implementation in `nacl_syscall_common.c` so that it calls a new fork RPC in `lind_platform.c` which transmits parent and child cage ID's.
3. Parse the RPC on the Repy side and handle that properly. The best way would be to change the flow so we can set our process "context" by cage ID. Forking, and copying FD's will have to be handled somewhat independently.

**Comment by rennergade** [comment]
I've updated the Lind RPC protocol so NaCl transmits cage-id on the `native_client` branch `nacl-rpc-cage` and @jesings has implemented a closure system to separate calls by cage on `nacl_repy`'s `cageid` branch.

The framework for creating an RPC call for `fork()` has been added on several `forkrpc` branches in `Lind_Glibc`, `native_client`, and `lind_project.

We need to add the FD table replication on fork to SafePOSIX via Repy. We'll have the call

1. Take both parent and child cage-id's as parameters.
2. Duplicate the parent's file table, and add the duplicate to the main dictionary of filetables by cage id.
